tree/dataframe/test/dataframe_regression.cxx

```diff
@@ -40,6 +40,28 @@ void FillTree(const char *filename, const char *treeName, int nevents = 0)
    f.Close();
 }
 
+// https://github.com/root-project/root/issues/11207
+TEST(RDFRegressionTests, AliasAndSubBranches)
+{
+   TTree t("t", "t");
+   auto x = std::make_pair(42, 84);
+
+   // This is a weird way to create a TTree, but we just need to mock a TTree with branch
+   // "topbranch.something" where `something` must not be a data member of the type
+   // of "topbranch" (otherwise things "happen" to work due to the order in which we do substitutions in RDF, see
+   // below).
+   t.Branch("topbranch", &x);
+   t.Branch("topbranch", &x, "a/I:b/I");
+   t.Fill();
+   t.Fill();
+   t.Fill();
+
+   auto df = ROOT::RDataFrame(t).Alias("alias", "topbranch");
+   // Here, before the fix for #11207 we transformed `"alias.a == 42"` into
```

Comment:
Can you also add what is the expected value of the lambda after the commit?

Reply:
done